### PR TITLE
enable setting CONFIG & CACHE filenames via ENV vars

### PR DESCRIPTION
**Pull request by handcode**
this PR consists of 2 changes:

- 70e0efcf3015302160356a788214a5ed7ae7a6f8
  fixes the 'each() function is deprecated' issue #16

- 8bc35aba941b5f31b29264af9dd7942e2c5d213d
  allows setting the two file names for `CONFIGFILE` and `CACHEFILE` using environment variables `PHPTIDY_CONFIGFILE`and `PHPTIDY_CACHEFILE`

  With the fixed filenames we often get `Warning: file_put_contents(./.phptidy-cache): failed to open stream: Permission denied in...` errors when calling phptidy.php in scripts whose $PWD is in directories where they do not have write access.

### CODE REVIEW

**Comment by cmrcx** [comment]
With your replacement for the each() one of the tests fails, because then the reset() does not work anymore. But I found a solution now.

**Comment by cmrcx** [comment]
I don't understand your explanation about the config and cache locations. Why don't you just change to the directory of the project you want to tidy up?

**Comment by handcode** [comment]
Sorry I didn't consider the reset. Great that you found another solution without each().

To the configuration and cache files: We use phptidy in a code generator inside docker containers to clean up dynamically generated files in a tmp directory. 

The phptidy.conf comes from a read-only directory (from the base image). So it would be useful for us if the hardcoded paths for config and cache files could be overwritten and separated from each other.

To cut a long story short, it's all about flexibility.

**Comment by cmrcx** [comment]
To specify the config file location you can use --config.

I would like to understand the reason for moving the cache file. If you have no write permission to the project directory, how should phptidy work there anyway? The cache file is not the only file that is written.

**Comment by handcode** [comment]
Rebased PR (removed "fix each()" commits) due to much better (and working) solution from @cmrcx see: 5418b60203b37c306814659580d244aa20712385

**Comment by handcode** [comment]
First of all thank you for your interest in our usecase.
I'll try to explain in more detail what our "problem" is.

As I said, we use phptidy within a rather complex code generator.

The generator reads the database schema and create appropriate Models,
Controllers and CRUD views. For that, the generator needs quite
complex configurations, a bunch of helper classes, etc., depending on
the project.

Within the entire process, phptidy is one of many commands and is
triggered as shell command from within PHP.

To be able to load all necessary configs, classes etc. the working
directory for the generator has to be the project root, which can
be read-only.

The target directory for generated files is of course writable,
but cannot be the working directory for the generator (see above).

With the current hardcoded paths we would have to switch back and
forth between the directories for each phptidy call.

Of course this is feasible, but in my opinion it would be much easier
and more flexible if you could simply set the path for the cache file
at runtime.

If you are bothered by the ENV variables, we could of course implement this with a new --cache option.

**Comment by cmrcx** [comment]
Well, phptidy is written for running in the directory of the project to be made tidy. Abandoning this premise would make everything more complicated, especially the documentation. You have to consider that at all file name variables and the @see tags. And it should work on all operating systems.

I could make the cache file location configurable in the config file, but I'm not convinced that this makes any sense. Having the cache file in the project directory seems to be the best solution. Anywhere else could be confusing about to which project the file belongs. If you just would like to set the cache file path to the project directory, maybe a command line option for the project directory would be more appropriate. But I think this is still more complicated than just switching the directories. I guess it is quite common for Unix commands to work on the current directory. Switching the directory is pretty easy, for example: 
$ cd generated_files/ ; phptidy.php replace ; cd -
$ ( cd generated_files/ && phptidy.php replace )

**Comment by handcode** [comment]
Hardcoded paths that cannot be changed via configurations are in my
opinion almost never a good solution, as different users will always
have different use cases in which they would like to adjust the paths.

But anyway, that's your project and so of course it's your
decision. So just close this PR if you like, we could work with a fork
which we patch the way we need it.

Thanks for your work!

PS: I know how to change directories, that never was my problem ;-)

**Comment by cmrcx** [comment]
I added some tests for the @see tags. They show, that if phptidy is called from a different directory, no @see tags are added, because the includes can not be found. And the default file docblock headlines show the wrong path.

Anyway, thanks for your appreciation!
